app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
1. the tasks would specify the implementation to be used for the input/output object, to achieve the extendability.
I assume one could define a default implementation in configuration. For example, the active record serializer/deserializer should be generic enough for majority of the Rails project.
2. I'm not sure I understand this part. What would be an example of dynflow caring about how the recipient task handles the output
3. yes, this part would stay the same as it's right now
4. no, the only constraint would be, that the `input/output` serializers know how to serialized/deserialize the objects to/from database.

Do this answers fit into you assumptions?